test/pre-commit-snapshots.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# This script is used as a pre-commit hook to run test for changed lua files.
+# It mostly just maps source lua files to their integration test
+#
+# Arguments:
+#   - $@: Changed lua sources
+
+#######################################
+# Variables
+#######################################
+
+# This is the directory where test scripts are located. It happens to be the
+# same as this script's, and it also happens to be inside the git repository
+TESTS_DIR="$(dirname "${BASH_SOURCE[0]}" | xargs readlink -f)"
+
+# Colored output
+RESET='\e[0m'
+GREEN='\e[92m'
+
+########################################
+# Main
+########################################
+
+basename -a "$@" | while read -r CHANGED_LUA_FILE; do
+    TEST_FILE="$TESTS_DIR/${CHANGED_LUA_FILE%%.lua}.spec.lua"
+    if [ -f "$TEST_FILE" ]; then
+        echo "$TEST_FILE"
+    else
+        echo -en >&2 "${GREEN}[TEST]${RESET} No snapshot found for "
+        echo -e >&2 "${GREEN}$CHANGED_LUA_FILE${RESET}"
+    fi
+done | xargs bash "$TESTS_DIR/regression-test.sh"
```

Comment:
This script _always_ pass no filename to `regression-test.sh`, thus having it run on all the files. I'm not a bash wizard so I'm not sure of why, but this is definitely unexpected.

Reply:
As far as my testing goes, it does pass file names. That's what `xargs` and `echo` in the while loop are for.

Reply:
I get this output, which actually does look like it's passing something, but then there's a bug somewhere because it's running all the tests regardless.
```bash
$ bash test/pre-commit-snapshots.sh Poké-data.lua
[TEST] No snapshot found for Poké-data.lua
[TEST] Running AbbrLink.spec.lua@current
[TEST] Running Abillist-allpokes.spec.lua@current
[TEST] Running Abillist.spec.lua@current
[TEST] Running Blackabbrev.spec.lua@current
[TEST] Running Colorabbrev.spec.lua@current
...
```

Reply:
Yeah, it doesn't work in that commit. In https://github.com/pokemoncentral/wiki-lua-modules/commit/91d4a43d6aa9f3f5e0a8aefe9f3528af63e4c921 it works though, I don't know what I did to fix it. :confused: